File: ecommerce-rest-api/src/main/java/com/example/ecommerce/service/OrderItemService.java
package com.example.ecommerce.service;

import com.example.ecommerce.dto.orderitem.OrderItemRequestDto;
import com.example.ecommerce.dto.orderitem.OrderItemResponseDto;

import java.util.List;

public interface OrderItemService {

    // Create
    OrderItemResponseDto create(
            OrderItemRequestDto request
    );

    // Get by ID
    OrderItemResponseDto getById(
            Long id
    );

    // Get all
    List<OrderItemResponseDto> getAll();

    // Update
    OrderItemResponseDto update(
            Long id,
            OrderItemRequestDto request
    );

    // Delete
    void delete(
            Long id
    );

    // Get by order ID
    List<OrderItemResponseDto> getByOrderId(
            Long orderId
    );
}

File: ecommerce-react-ui/src/services/orderItemService.js
import api from "./api";

const orderItemService = {
    getAll: async () => {
        const response = await api.get("/order-items");
        return response.data;
    },

    getById: async (id) => {
        const response = await api.get(`/order-items/${id}`);
        return response.data;
    },

    getByOrder: async (orderId) => {
        const response = await api.get(
            `/order-items/order/${orderId}`
        );
        return response.data;
    },

    create: async (orderItem) => {
        const response = await api.post(
            "/order-items",
            orderItem
        );
        return response.data;
    },

    update: async (id, orderItem) => {
        const response = await api.put(
            `/order-items/${id}`,
            orderItem
        );
        return response.data;
    },

    remove: async (id) => {
        const response = await api.delete(
            `/order-items/${id}`
        );
        return response.data;
    },
};

export default orderItemService;